sanitizers/sql.go

```diff
@@ -0,0 +1,94 @@
+package sanitizers
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"regexp"
+)
+
+var syntaxErrors = []*regexp.Regexp{
+	regexp.MustCompile(`\S+ ERROR 1064 \(42000\): You have an error in your SQL syntax.*`), // MySQL error message
+	regexp.MustCompile(`\S+ ERROR: syntax error at or near .* \(SQLSTATE 42601\)`),         // PostgreSQL error message
+}
+
+func ExecDB(fakPC int, db *sql.DB, query string, args ...any) (sql.Result, error) {
+	result, err := db.Exec(query, args...)
+	if isSyntaxError(err) {
+		panic(fmt.Sprintf("GoSan: SQL Injection: query %s, args %s", query, fmt.Sprint(args)))
+	}
+	return result, err
+}
+
+func ExecContextDB(fakPC int, db *sql.DB, ctx context.Context, query string, args ...any) (sql.Result, error) {
+	result, err := db.ExecContext(ctx, query, args...)
+	if isSyntaxError(err) {
+		panic(fmt.Sprintf("GoSan: SQL Injection: query %s, args %s", query, fmt.Sprint(args)))
+	}
+	return result, err
+}
+
+func QueryDB(fakePC int, db *sql.DB, query string, args ...any) (*sql.Rows, error) {
```

Comment:
Add `QueryRow` and `QueryRowContext`?

Reply:
Are there use cases where one would execute `QueryRow,` or  `QueryRowContext` is used without performing a scan afterward? These functions do not return an error, and then we should parse the query to check whether it is syntactically valid